docs/book/provider_implementations/building_running_and_testing.md

```diff
@@ -6,24 +6,24 @@ Apply the following patch:
 
 ```bash
 diff --git a/Makefile b/Makefile
-index ac12c7e..9b4f945 100644
+index ac12c7e..cf44312 100644
 --- a/Makefile
 +++ b/Makefile
-@@ -22,12 +22,14 @@ install: manifests
+@@ -22,12 +22,15 @@ install: manifests
  
  # Deploy controller in the configured Kubernetes cluster in ~/.kube/config
  deploy: manifests
--       kubectl apply -f config/crds
--       kustomize build config/default | kubectl apply -f -
-+       cat provider-components.yaml | kubectl apply -f -
+-	kubectl apply -f config/crds
+-	kustomize build config/default | kubectl apply -f -
++	cat provider-components.yaml | kubectl apply -f -
  
  # Generate manifests e.g. CRD, RBAC etc.
  manifests:
--       go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go all
-+       go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go crd
-+       kustomize build config/default/ > provider-components.yaml
-+       echo "---" >> provider-components.yaml
-+       kustomize build vendor/sigs.k8s.io/cluster-api/config/default/ >> provider-components.yaml
+ 	go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go all
++	sed -i'' -e 's@^- manager_auth_proxy_patch.yaml.*@#&@' config/default/kustomization.yaml
```

Comment:
Any particular reason here to use `@` as the delimiter here instead of the more common `/`?

Reply:
@detiber  No particular reason, just to keep the style consistent with kubebuilder generated files, such as https://github.com/kubernetes-sigs/cluster-api/blob/master/Makefile.

Reply:
The `-i` option to `sed` is OS specific (cf. https://stackoverflow.com/questions/2320564/i-need-my-sed-i-command-for-in-place-editing-to-work-with-both-gnu-sed-and-bsd). We might add a comment, but fixing this for real might be best done by moving away from using the Makefile to construct `provider-components.yaml`.

Reply:
I believe @figo is the one who found a decent workaround and implemented it here: https://github.com/kubernetes-sigs/cluster-api/blob/master/Makefile#L112

Reply:
agree, will be nice to include the workaround `sed -i.tmp`, it works cross MacOs/Linux.